src/drivers/intel/dmic/dmic_nhlt.c

```diff
@@ -272,22 +278,24 @@ int dmic_set_config_nhlt(struct dai *dai, void *spec_config)
 	/* Write the FIFO control registers. The clear/set of bits is the same for
 	 * all DMIC_HW_VERSION
 	 */
+	/* Clear TIE, SIP, FCI, set FINIT, the rest of bits as such */
+	val = (out_control[dai->index] &
+		~(OUTCONTROL0_TIE_BIT | OUTCONTROL0_SIP_BIT | OUTCONTROL0_FCI_BIT)) |
+		OUTCONTROL0_FINIT_BIT;
 	if (dai->index == 0) {
-		/* Clear TIE, SIP, FCI, set FINIT, the rest of bits as such */
-		val = (out_control[0] &
-			~(OUTCONTROL0_TIE_BIT | OUTCONTROL0_SIP_BIT | OUTCONTROL0_FCI_BIT)) |
-			OUTCONTROL0_FINIT_BIT;
 		dai_write(dai, OUTCONTROL0, val);
-		dai_dbg(dai, "dmic_set_config_nhlt(): OUTCONTROL0 = %08x", val);
+		dai_info(dai, "dmic_set_config_nhlt(): OUTCONTROL0 = %08x", val);
 	} else {
-		/* Clear TIE, SIP, FCI, set FINIT, the rest of bits as such */
-		val = (out_control[1] &
-			~(OUTCONTROL1_TIE_BIT | OUTCONTROL1_SIP_BIT | OUTCONTROL1_FCI_BIT)) |
-			OUTCONTROL1_FINIT_BIT;
 		dai_write(dai, OUTCONTROL1, val);
-		dai_dbg(dai, "dmic_set_config_nhlt(): OUTCONTROL1 = %08x", val);
+		dai_info(dai, "dmic_set_config_nhlt(): OUTCONTROL1 = %08x", val);
 	}
```

Comment:
You can move the print out from the two loops to:
```c
if (dai->index == 0)
	dai_write(dai, OUTCONTROL0, val);
else
	dai_write(dai, OUTCONTROL1, val);

dai_info(dai, "dmic_set_config_nhlt(): OUTCONTROL%d = %08x", dai->index, val);
```

Reply:
Done!